### ISSUE TITLE
[data grid] Allow to control subscribe event priority

### ISSUE THREAD

**Issue by httpete** [mui/mui-x#3015]
### Summary 💡

I have the need to resize a column programatically.  I am using the version 5@next.
For keyboard a11y, when a column is focused, pressing CTRL+ left or right arrow should be able to resize the column.

I can hook onto the subscribeEvent('columnHeaderKeyDown') quite nicely, (although I found that the useGridApiEvent hook to not work)

But I can't see how I can resize a column programatically through an API call.

### Examples 🌈

````
 useEffect(() => {
    return apiRef.current.subscribeEvent(
      'columnHeaderKeyDown',
      (d, e, apiPer) => {
        if(e.ctrlKey == true && e.code == 'ArrowRight'){
          //I just want to do column.setWidth('250px') for example
        }
      }
    );
  }, [apiRef]);
````
Here is how I did it in ag-grid's api:
````
   const currentWidth = column.getActualWidth();

    enum Operators {
      ADD = '+',
      SUB = '-',
    }

    const operators = {
      [Operators.ADD]: function (a: number, b: number) {
        return a + b;
      },
      [Operators.SUB]: function (a: number, b: number) {
        return a - b;
      },
    } as const;

    let operator = Operators.ADD;
    if (key == arrowLeftKey) {
      operator = Operators.SUB;
    }
    columnApi.setColumnWidth(column, operators[operator](currentWidth, 10));
````
### Motivation 🔦

 For keyboard a11y, when a column is focused, pressing CTRL+ left or right arrow should be able to resize the column.

**Comment by m4theushw** [comment]
What is the conclusion here? Should we update https://mui.com/components/data-grid/events/#subscribing-to-events mentioning what is `GridSubscribeEventOptions`. I consider this API public. If users can have priority on a callback defined via `onXXX` props, then they should be allowed to get the same outcome by using `subscribeEvent`.

**Comment by flaviendelangle** [comment]
I think we should make the priority option public. 
`isFirst` is not fully representative of what the option is doing since if you have other `isFirst` listeners added after, they will be called before.
Do you think the name is explicit enough ?
If not, we can find a better name before making it public.

**Comment by flaviendelangle** [comment]
The PR should also update this paragraph which is currently only talking about the events with an `onXXX` prop. https://mui.com/components/data-grid/events/#disabling-the-default-behavior

**Comment by httpete** [comment]
any progress in this regard? We are finding as datagrid/pro customers without this ability to customize some behavior we are blocked.

**Comment by flaviendelangle** [comment]
@cherniavskii I'm adding you in the loop here :+1:

**Comment by michelengelen** [comment]
Thanks for the report. The underlying need is covered nowadays: you can resize columns programmatically with `apiRef.current.setColumnWidth(field, width)` or `apiRef.current.autosizeColumns()`. The `isFirst` subscription option discussed above no longer exists, and generic control over event subscription priority hasn't gained traction since 2021, so it's not on the roadmap.

Closing as not planned. If you have a concrete use case that still requires priority control, please open a new issue describing it.
